src/components/show-products/index.js

```diff
@@ -43,27 +67,46 @@ export const ShowProducts = () => {
 
   return (
     <ContentContainer>
+      <form className="filter-form">
+        <label htmlFor="filter">
+          Filter items
+          <input
+            className="input-item"
+            type="search"
+            id="filter"
+            name="filter"
+            aria-label="Search through list content"
+            placeholder="Start typing here..."
+            value={searchTerm}
+            onChange={(event) => setSearchTerm(event.target.value)}
+          />
+        </label>
+      </form>
       <ul>
-        {products?.map(({ id, productName, lastPurchaseDate, timeFrame }) => (
-          <li className="checkbox-item" key={id}>
-            <label htmlFor={id} className="checkbox-label">
-              <input
-                type="checkbox"
-                id={id}
-                name={productName}
-                onChange={() => handleOnChange(id)}
-                checked={
-                  lastPurchaseDate && compareDates(lastPurchaseDate.toDate())
-                }
-                ariaLabel={timeFrames[timeFrame]}
-              />
-              <span
-                className={`checkmark checkbox-timeFrame-${timeFrame}`}
-              ></span>
-              <span className="checkbox-name">{productName}</span>
-            </label>
-          </li>
-        ))}
+        {list.length > 0 ? (
+          list.map(({ id, productName, lastPurchaseDate, timeFrame }) => (
```

Comment:
@yanurd Since we do the empty list check here, we don't need the optinal chaining, but your previous comment is correct.

Reply:
Oh! I totally missed that line, I wonder what mentors would suggest, even though great work !